python/cudf_polars/cudf_polars/experimental/parallel.py

```diff
@@ -234,3 +255,61 @@ def evaluate_dask(ir: IR) -> DataFrame:
 
     graph, key = task_graph(ir, partition_info)
     return get(graph, key)
+
+
+def _concat(dfs: Sequence[DataFrame]) -> DataFrame:
+    # Concatenate a sequence of DataFrames vertically
+    return Union.do_evaluate(None, *dfs)
+
+
+##
+## Union
+##
+
+
+@lower_ir_node.register(Union)
+def _(
+    ir: Union, rec: LowerIRTransformer
+) -> tuple[IR, MutableMapping[IR, PartitionInfo]]:
+    # zlice must be None
+    if ir.zlice is not None:
+        return rec.state["default_mapper"](ir)  # pragma: no cover
```

Comment:
Right - this line is just a punt. Hopefully that's okay.

Reply:
Ah, so to answer my question about why we need the "single partition" handler, we want something to punt with that will fail if children have more than one partition. So that we don't have to handle that here?

We could do something like:

```
children, partitions = zip(...)
if zlice is not None:
    if any(p[c].count > 1 for p, c in zip(children, partitions)):
        raise ...
    return ir.reconstruct(children), ...
```
I suppose?